clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -547,6 +551,8 @@ static KafkaAdminClient createInternal(
             MetricsContext metricsContext = new KafkaMetricsContext(JMX_PREFIX,
                     config.originalsWithPrefix(CommonClientConfigs.METRICS_CONTEXT_PREFIX));
             metrics = new Metrics(metricConfig, reporters, time, metricsContext);
+            clientTelemetryReporter = CommonClientConfigs.telemetryReporter(clientId, config);
+            clientTelemetryReporter.ifPresent(telemetryReporter -> telemetryReporter.contextChange(metricsContext));
```

Comment:
Though it's good to have the integartion of KIP-714 with admin client but I left it last time because of adhoc connections by Admin client, this will establish telemetry connection to send metrics, do we require them? Secondly, does admin client has any metric of it's own in Kafka? At the time of KIP-714, I didn't see any relevant metric for admin client. Last month we added https://issues.apache.org/jira/browse/KAFKA-17239 to record node latency which was related to https://issues.apache.org/jira/browse/KAFKA-17230. But again should we have KIP-714 pipeline for this 1 admin metric, I am not sure.

Hence should we register additional metrics here? Seems more of an additional overhead for adhoc requests. Else if we want external applications to register additional metrics in Admin client anyways then we might want to consider only enabling client telemetry reporter if additional metrics are registered. Wdyt?

Reply:
That's a good point, I'll take a look at doing that.

Reply:
I'm leaning toward not doing this, as I agree with @mjsax. Plus, there's additional complexity in the case where metrics are unregistered. Should that trigger the mirror action of disabling metrics pushing? 

>Btw: I was just looking into the consumer/producer code, and it seem there we just create the reporter and add to reporters list before new Metrics is called -- this way, metricsContext will be "added" automatically to the reporter. Should simplify the code a little bit?

I'm not sure I follow this comment.  Are you referring to the reporters we create in Kafka Streams to act as proxies for passing the metrics to the clients and suggesting we create them in the consumer/producer clients themselves?   
Those `Reporter` instances are specific to Kafka Streams only.  Also, the `Metrics` API provides an `addReporter` method, so it seems it was designed to add a reporter after instantiating the `Metrics` object.

Reply:
So the only metric I think admin client emits is `node latency` metric, which also got recently added as regression. Though I understand that KS wants to send application metrics as part of admin client but we will be adding an additional overhead with short lived admin client operations. I am not very aligned regarding with KIP-714 integration in admin client. @AndrewJSchofield wdyt?

Reply:
@apoorvmittal10 I spoke offline with @mjsax and what we've come up with is that we'll disable metrics in the AdminClient by default (I'll update KIP-1076 with this detail).  KafkaStreams will explicitly enable the metrics push for the AdminClient, avoiding any additional overhead for non-KafkaStreams usage.  WDYT?

Reply:
That sounds fair to me. For my knowledge, does admin client in streams application has a long running connection where metrics will be emitted? My concern is with connections are adhoc and we do create too many admin client short lived connections then it might impact the connection cache i.e. connection cache keep the client instance for maximum of push interval * 3 ms hence if too many admin client connections created in short span then it might hurt the performance.

Reply:
Sounds good to me too.

Reply:
Kafka Streams uses the admin client to also send "delete record" request for repartition topics. For this case, the connection should be long lived, as we send a "delete record" request after each commit by default (it's actually configurable).

For the (more rare) case that there is no repartition topic, it could be a slightly different story, but IIRC, @bbejeck did some tests already, verifying that the connection is not closed by long lived, even for this case. So I think we are good?

We could run more test before the 4.0 release to double check, and if we really find an issue, we could still update the KIP, to let KS use different connection timeout default, to make sure that the connection will be long lived.